gtest/src/manager.rs

```diff
@@ -30,17 +30,70 @@ use gear_core::{
 use std::collections::{BTreeMap, VecDeque};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-#[derive(Clone, Debug)]
-pub(crate) enum ProgramState {
-    Initialized,
-    Uninitialized(Option<MessageId>),
-    FailedInitialization,
+#[derive(Debug)]
+pub(crate) enum Actor {
+    Active(ActiveProgram, ProgramState, u128),
+    Dormant,
+}
+
+impl Actor {
+    pub(crate) fn new_active(active_prog: ActiveProgram) -> Self {
+        Actor::Active(active_prog, ProgramState::Uninitialized(None), 0)
+    }
+
+    fn get_state_mut(&mut self) -> Option<&mut ProgramState> {
+        match self {
+            Actor::Active(_, state, _) => Some(state),
+            _ => None,
+        }
+    }
+
+    fn try_into_core_prog_mut(&mut self) -> Result<&mut CoreProgram, ()> {
```

Comment:
This and other methods below with `try_into` semantic aren't correct in rust world. `into` in any representation means self consume. I would go here with `as_mut_core_prog() -> Option<mut ref>`, but they look redundant for me anyway

Reply:
`as_` by convention is cost free "borrow -> borrow" operation. I will try to name it closer to the domain